### DISCUSSION TITLE
Proposal: <podcast:valueTimeSplit> tag

### DISCUSSION THREAD

**Discussion by agates** [Podcastindex-org/podcast-namespace#512] [Tag Proposal]
# Proposal: &lt;podcast:valueTimeSplit&gt; tag

The intent of this proposal is to provide a tag that allows different value splits for a certain period of time. It is a combination of the concept of `<podcast:soundbite>` and `<podcast:remoteItem>` where a start time and a duration is supplied with alternative value recipients. The alternative value recipients are not required to be remote, as the recipients may not have an RSS feed/item of their own to reference.

This is suggested as an alternative to a chapter-based approach to simplify the implementation and use existing XML tags.

## Example referencing external music

The following shows an example referencing two different external songs. The first is played in full starting at 1 minute into the podcast. The second is played 5 minutes and 30 seconds into the podcast for a duration of 53 seconds, starting at 2 minutes and 54 seconds of the song featuring a wicked guitar solo.

```xml
<rss xmlns:podcast="https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md" version="2.0">
    <channel>
    <title>Metal Showcase</title>
    <description>This is a podcast featuring badass v4v metal music</description>
    <link>https://example.com/rss-metal-showcase.xml</link>
    <docs>http://blogs.law.harvard.edu/tech/rss</docs>
    <language>en</language>
    <generator>Alecks Gates wrote this by hand in Joplin</generator>
    <pubDate>Fri, 21 Apr 2023 18:56:30 -0500</pubDate>
    <lastBuildDate>Fri, 21 Apr 2023 18:56:30 -0500</lastBuildDate>
    <podcast:medium>podcast</podcast:medium>
    <item>
        <title>Special interview with Torcon VII</title>
        <otherTagsHere>...</otherTagsHere>
        <podcast:value type="lightning" method="keysend">
            <podcast:valueRecipient name="Alice (Podcaster)" type="node" address="02d5c1bf8b940dc9cadca86d1b0a3c37fbe39cee4c7e839e33bef9174531d27f52" split="95" />
            <podcast:valueRecipient name="Hosting Provider" type="node" address="03ae9f91a0cb8ff43840e3c322c4c61f019d8c1c3cea15a25cfc425ac605e61a4a" split="5" fee="true" />

            <podcast:valueTimeSplit startTime="60" duration="237" remotePercentage="95">
                <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#1" feedGuid="a94f5cc9-8c58-55fc-91fe-a324087a655b" medium="music" />
            </podcast:valueTimeSplit>

            <podcast:valueTimeSplit startTime="330" duration="53" remoteStartTime="174" remotePercentage="95">
                <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#3" feedGuid="a94f5cc9-8c58-55fc-91fe-a324087a655b" medium="music" />
            </podcast:valueTimeSplit>
        </podcast:value>
    </item>
    </channel>
</rss>
```

## Example referencing different value recipients

The following shows an example different value recipients for a two specific segments of a show

```xml
<rss xmlns:podcast="https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md" version="2.0">
    <channel>
    <title>Cool Pod</title>
    <description>This is a cool pod</description>
    <link>https://example.com/rss-cool-pod.xml</link>
    <docs>http://blogs.law.harvard.edu/tech/rss</docs>
    <language>en</language>
    <generator>Alecks Gates wrote this by hand in Joplin</generator>
    <pubDate>Fri, 21 Apr 2023 18:56:30 -0500</pubDate>
    <lastBuildDate>Fri, 21 Apr 2023 18:56:30 -0500</lastBuildDate>
    <podcast:medium>podcast</podcast:medium>
    <item>
        <title>Adam Hates the word "pod" (and I do, too)</title>
        <otherTagsHere>...</otherTagsHere>
        <podcast:value type="lightning" method="keysend">
            <podcast:valueRecipient name="Alice (Podcaster)" type="node" address="02d5c1bf8b940dc9cadca86d1b0a3c37fbe39cee4c7e839e33bef9174531d27f52" split="95" />
            <podcast:valueRecipient name="Hosting Provider" type="node" address="03ae9f91a0cb8ff43840e3c322c4c61f019d8c1c3cea15a25cfc425ac605e61a4a" split="5" fee="true" />

            <podcast:valueTimeSplit startTime="60" duration="300">
                <podcast:valueRecipient name="Alice (Podcaster)" type="node" address="02d5c1bf8b940dc9cadca86d1b0a3c37fbe39cee4c7e839e33bef9174531d27f52" split="85" />
                <podcast:valueRecipient name="Jimbob (Guest)" type="node" address="02dd306e68c46681aa21d88a436fb35355a8579dd30201581cefa17cb179fc4c15" split="10" />
            </podcast:valueTimeSplit>

            <podcast:valueTimeSplit startTime="360" duration="240">
                <podcast:valueRecipient name="Alice (Podcaster)" type="node" address="02d5c1bf8b940dc9cadca86d1b0a3c37fbe39cee4c7e839e33bef9174531d27f52" split="85" />
                <podcast:valueRecipient name="Bobjim (Guest)" type="node" address="032f4ffbbafffbe51726ad3c164a3d0d37ec27bc67b29a159b0f49ae8ac21b8508" split="10" />
            </podcast:valueTimeSplit>
        </podcast:value>
    </item>
    </channel>
</rss>
```

## Definition

`<podcast:valueTimeSplit>`

The `<podcast:valueTimeSplit>` tag allows time-based changes of  value recipients in a given value block for an item.

This can either contain one or more `<podcast:valueRecipient>` tags *or* exactly one `<podcast:remoteItem>` tag. If a `<podcast:remoteItem>` tag is present, the `remotePercentage` attribute can be defined to control how much the remote value block's `<podcast:valueRecipient>` tags will receive as a whole on top of the default, non-fee `<podcast:valueRecipient>` tags.

If the remote value block contains any `<podcast:valueTimeSplit>` tags, they should be ignored even if the `remoteStartTime` indicates it's in a position that would invoke them. When referencing a remote value block, only the root level block splits should be used and any `<podcast:valueTimeSplit>` children are to be ignored.

Fees from the default `<podcast:valueRecipient>` tags should remain to be calculated before anything is taken out from `<podcast:valueTimeSplit>`.

### Parent

`<podcast:value>`

### Count

Multiple

### Attributes

- `startTime` (required) - The time in the current item where the value split begins
- `duration` (required) - How long the value split lasts
- `remoteStartTime` (optional) - The time in the remote item where the value split begins.  Allows the timestamp to be set correctly in value metadata.  If not defined, defaults to 0.
- `remotePercentage` (optional) - The percentage of the payment the remote recipients will receive if a `<podcast:remoteItem>` is present.  If not defined, defaults to 100.  If the value is less than 0, 0 is assumed.  If the value is greater than 100, 100 is assumed.

**Comment by adamc199** [comment]
I like this a lot. It adds a few thing I had not considered.

**Comment by thebells1111** [comment]
I'm thinking about the TLV that gets sent to the remoteSplit. It seems like we could add a clipped by podcast/episode in the TLV so the remoteSplit knows who is clipping their episode. That's easy enough to get from the main channel/item. Also, the clipped podcast and episode are easy enough to get from the two guids, but some way to have a timestamp of where the clip is occurring in the remoteItem might be nice to be able to include in the remoteSplit, so the clipped podcaster can see in their TLV record exactly what was being clipped from their episode.

**Reply by agates** [reply]
> I'm thinking about the TLV that gets sent to the remoteSplit. It seems like we could add a clipped by podcast/episode in the TLV so the remoteSplit knows who is clipping their episode. That's easy enough to get from the main channel/item. Also, the clipped podcast and episode are easy enough to get from the two guids, but some way to have a timestamp of where the clip is occurring in the remoteItem might be nice to be able to include in the remoteSplit, so the clipped podcaster can see in their TLV record exactly what was being clipped from their episode.

I actually had a `remoteStartTime` attribute in my original spec writeup but removed it.  Would that provide the functionality you're describing?

**Reply by thebells1111** [reply]
> I actually had a `remoteStartTime` attribute in my original spec writeup but removed it. Would that provide the functionality you're describing?

Yeah, that would work.

**Comment by daveajones** [comment]
What is the possibility of remote time split inception with this approach?  Is it something we need to be worried about?  If so, we can just put wording in like:  "If the value block from the remote item also contains a `<podcast:valueTimeSplit>`, that is to be ignored even if the `remoteStartTime` currently indicates it's in a position that would invoke it.  When referencing a remote value block, only the root level block splits should be used and any `<podcast:valueTimeSplit>` children are to be ignored". 

That's hard wording. I hope I'm making sense.

**Reply by agates** [reply]
That makes sense, yes.  I will add that.

**Comment by thomasrynne** [comment]
As described, clips that don't span a 1 minute marker would not receive any streaming sats. For example, the 53 second clip above would not receive any streaming sats if the start time was "305". If clips are likely to be more than 1 minute this doesn't matter. (I haven't implemented streaming sats myself yet so I may have misunderstood the podcast:value spec)

**Reply by agates** [reply]
There is no actual mandatory requirement to have minute-by-minute payments, that has just how people have been implementing it for statistical purposes.  It's very limited for reasons you described and I think apps should make sure to account for this.

In the future I want to push people to have less payments overall and send batch timestamp metadata in TLVs instead of having individual payments, which would help address the issue by thinking of listening/watch statistics as metadata instead of number of payments.

**Reply by thomasrynne** [reply]
The spec does say [The "time interval" for calculating payments is always 1 minute](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md#payment-intervals) which I understood to mean on the minute boundary but I guess that can be interpreted in different ways

**Reply by agates** [reply]
Nothing enforces clients to adhere to bad specifications, at the end of the day they are only suggestions :)

I provided an example of how we can get passed in [this Mastodon thread](https://podcastindex.social/@StevenB/110316627615432258)

**Reply by daveajones** [reply]
> The spec does say [The "time interval" for calculating payments is always 1 minute](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md#payment-intervals) which I understood to mean on the minute boundary but I guess that can be interpreted in different ways

The "1 minute" is a sliding window instead of a boundary.  Some podcasts could actually be less than 1 minute (like maybe a music track) so it's not necessary to hit a certain minute boundary.

**Comment by samsethi** [comment]
IDEA: If a podcast supports music chapters like Boostergram Ball. Then each track has a unique itemGUID with a start time and duration.  Each music track is really just a link to an external podcast episode/track. 

<img width="1171" alt="Sovereign_Feeds" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/6541efe7-edd4-4007-8c7c-c5981e41b9d2">


Podfans already allows users to create episode-level playlists from podcasts using the itemGUID. This also means Podfans can now reward the creator of the playlist with a small 1% split if any of the episodes in their playlist is shared or played. #V4V

Example of a Phase 7 tag? 
<podcast:playlist>
    <podcast:remoteItem feedGuid="29cdca4a-32d8-56ba-b48b-09a011c5daa9" itemGuid="asdf089j0-ep240-20230510" />
    <podcast:remoteItem feedGuid="396d9ae0-da7e-5557-b894-b606231fa3ea" itemGuid="bsdf089j0-ep240-20230510"/>
    <podcast:remoteItem feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" itemGuid="csdf089j0-ep240-20230510" />
</podcast:playlist>

But what if we extend this remoteItem to the chapter level for a non-music podcast. We could also use startTime and Duration fields to link to a specific point in a chapter.  

<podcast:remoteItem
    feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810"
    itemGuid="asdf089j0-ep240-20230510"
    chapterGuid="bsaf043j0-sp260-50270619
/>

<img width="1318" alt="Podfans___sam__Sam_Sethi_" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/9a240f3a-61c7-44d1-aed6-e0fa9e73a9aa">

Of course I answered my own question.  We don't need a chapterGUID if we simply use the ValueTimeSplit start time and duration on the itemGUID where the medium is 'podcast' In fact we could contain multiple ValueTimeSplit tags in the RSS feed from multiple podcasts if we also use the feedGUID. 

 <podcast:valueTimeSplit startTime="60" duration="237" remotePercentage="95">
               <podcast:remoteItem itemGuid="https://podcastindex.org/podcast/4148683#1" feedGuid="a94f5cc9-8c58-55fc-91fe-a324087a655b" medium="podcast" />
            </podcast:valueTimeSplit>

The question now is what UI to give podcast creators so they can create PodRolls, simply by looking up the name of a podcast but storing it as a feedGUID in the RSS. We already have a search against the PodcastIndex for users.  We can adapt this search to make the action button 'add to podroll'

<img width="1298" alt="Podfans___Search_for_Podcasts" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/0ff6cd25-2815-4854-8ec4-14702cd6f7f4">

<podcast:podroll>
    <podcast:remoteItem feedGuid="29cdca4a-32d8-56ba-b48b-09a011c5daa9" />
    <podcast:remoteItem feedGuid="396d9ae0-da7e-5557-b894-b606231fa3ea" />
    <podcast:remoteItem feedGuid="917393e3-1b1e-5cef-ace4-edaa54e1f810" />
</podcast:podroll>

We already have a UI for users to create Playlists from one podcast or multiple podcasts. 

<img width="1254" alt="Cursor_and_Podfans___Shows___Podnews_Weekly_Review" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/afbde2fe-93c0-45ed-a61c-185febf29915">

 We could extend this UI to chapters by adding an action button next to chapters but we would still store the chapter playlist using the itemGUID and take the chapter time marker and duration between chapters. 

![Podfans___Shows___Telling_Media_s_Christiana_Brenton_and_Georgie_Holt__plus_C-18_in_Canada_and_Audioboom_s_results](https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/78a2313c-21d0-47c4-8ec7-f23d2e12e37b)

The question is would users want a chapter level playlist.  I can see the value in doing this with music playlists. Boostergram is storing music tracks/episodes as chapters. 

The other UI is for podcast creators. Sovreign Feeds is the only/best option right now. But how could other podcast apps offer creators a similar UI. Dave and Adam often play clips on their Podcasting 2.0 show.  What if those clips were stored in the RSS feed as ValueTimeSplits???

**Reply by agates** [reply]
This is the idea of [List Mediums](https://github.com/Podcastindex-org/podcast-namespace/discussions/480) with [`<podcast:remoteItem>`](https://github.com/Podcastindex-org/podcast-namespace/issues/346) where you can create RSS feeds as dedicated playlists and, optionally, cross-app clips -- and serve them publicly on behalf of your users.  That re-uses the `<podcast:soundbite>` tag.

We may need to re-evaluate value attribution on those scenarios, I could see a 1% split being reasonable.  Maybe that should be up to the app, however.  If users were able to set their own split in a playlist that would be easily abused.

If you wanted to create playlists all you would do is host the RSS feed and submit it to Podping -- I don't know if @daveajones consumes them yet in podcast index.

**Reply by thebells1111** [reply]
Since we're using remoteGuids to reference a feed, perhaps we need some sort of new <podcast:curation split='1'>tag  in the referenced feed where the podcaster decides how much they are willing to share with a curator. It can default to zero if not present.
